okio/src/jvmMain/kotlin/okio/HashingSink.kt

```diff
@@ -41,11 +41,21 @@ class HashingSink : ForwardingSink {
   private val messageDigest: MessageDigest?
   private val mac: Mac?
 
+  internal constructor(sink: Sink, digest: MessageDigest) : super(sink) {
+    this.messageDigest = digest
+    this.mac = null
+  }
+
   internal constructor(sink: Sink, algorithm: String) : super(sink) {
     this.messageDigest = MessageDigest.getInstance(algorithm)
```

Comment:
change this constructor to call through the new one?

Reply:
Yup, you're right, makes sense